### PR TITLE
added methods to project.js as well as more detailed search

### PR DESCRIPTION
**Pull request by fonzy2013**
Added createProject, deleteProject, getProjectProperties as well as changed to use GET instead of POST for search as it then can take expand and other parameters

### CODE REVIEW

File: api/search.js [hunk 1]
```diff
@@ -52,20 +52,22 @@ function SearchClient(jiraClient) {
     this.search = function (opts, callback) {
         var options = {
             uri: this.jiraClient.buildURL('/search'),
-            method: 'POST',
+            method: 'GET', // NB1. changed to GET to retrieve all properties when searching.
```
Comment (floralvikings) [inline api/search.js]: Per the [Jira API Documentation](https://docs.atlassian.com/jira/REST/latest/#api/2/search-search):

> GET vs POST: If the JQL query is too large to be encoded as a query param you should instead POST to this resource.

While I doubt many people are sending search queries long enough for this to be a problem, I'd rather not change it unless absolutely necessary.  What properties are available over `GET` that aren't over `POST`?

File: api/search.js [hunk 2]
```diff
@@ -52,20 +52,22 @@ function SearchClient(jiraClient) {
...
                 startAt: opts.startAt,
                 maxResults: opts.maxResults,
                 validateQuery: opts.validateQuery,
                 fields: opts.fields,
                 expand: opts.expand
-            }
+            }*/
         };
-
+        console.log2(options);
```
Comment (floralvikings) [inline api/search.js]: Looks like this was probably left in from testing

File: api/search.js (hunk 1 above)
Comment (fonzy2013) [inline api/search.js]: When I tested I didn't get any custom fields returned when using POST. While in GET I did. Probably a bug in jira-rest-api but I really need to have custom fields.

What do you say about having GET vs POST configurable as 
```
opts.method = 'GET'; //default 'POST'
```

File: api/search.js (hunk 2 above)
Comment (fonzy2013) [inline api/search.js]: True. I'll remove it.

File: api/search.js (hunk 1 above)
Comment (floralvikings) [inline api/search.js]: What a strange bug 😕 

I like the idea of making it configurable, let's go with that; add that functionality and remove the console log below, and I'll merge and publish.

Thanks!

Comment (fonzy2013) [inline api/search.js]: Thanks! Will do that. I'll do it this weekend right now I'm busy with other work.

**Comment by fonzy2013** [comment]
I have now finally gotten around to fix what I promised. It is tested and works.

**Comment by fonzy2013** [comment]
@floralvikings did you have a chance to look into this PR?

**Comment by floralvikings** [comment]
Yup, sorry for the delay, been swamped lately

**Pull request merged**